src/k8s/pkg/k8sd/controllers/csrsigning/reconcile_approve.go

```diff
@@ -5,14 +5,28 @@ import (
 	"fmt"
 
 	"github.com/canonical/k8s/pkg/log"
+	pkiutil "github.com/canonical/k8s/pkg/utils/pki"
 	certv1 "k8s.io/api/certificates/v1"
 	v1 "k8s.io/api/core/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
 func (r *csrSigningReconciler) reconcileAutoApprove(ctx context.Context, log log.Logger, csr *certv1.CertificateSigningRequest) (ctrl.Result, error) {
```

Comment:
Accept the private key as an argument, we have already fetched the cluster config once in the reconcile loop already https://github.com/canonical/k8s-snap/blob/2994b879eab39b782556f122a2c8512bf85e9aca/src/k8s/pkg/k8sd/controllers/csrsigning/reconcile.go#L56-L69

Reply:
Also, consider whether we need to account for a nil key or not.